src/hip/HIPStream.cpp

```diff
@@ -33,6 +33,17 @@ HIPStream<T>::HIPStream(const int ARRAY_SIZE, const int device_index)
     throw std::runtime_error(ss.str());
   }
 
+  // The array size must be divisible by total number of elements
```

Comment:
Why do we need to add this restriction? The CUDA version handles this case.

Reply:
I think you're right Tom, this is a hold-over from a previous version I think.

Reply:
@tomdeakin Apologies for the late reply --- I've gone ahead and pushed an additional commit removing this